src/asm/output.c

```diff
@@ -884,10 +854,8 @@ void out_PCRelByte(struct Expression *expr)
 	checksectionoverflow(1);
 
 	/* Always let the linker calculate the offset. */
-	if (nPass == 2) {
-		pCurrentSection->tData[nPC] = 0;
-		createpatch(PATCH_BYTE_JR, expr);
-	}
+	pCurrentSection->tData[nPC] = 0;
```

Comment:
This might be unrelated to this PR per-se, but why isn't `rpn_IsReloc()` being used here, like in `out_RelLong`?
As far as I understand, this will prevent outputting patches where only constants are used, and no labels. While this is less useful for relative jumps, I know for example `jr cc, @ + 1` will be used sometimes, and the comment above also mentions "Checking will be done to see if it is an absolute value in disguise.".

Reply:
Using `rpn_IsReloc()` wouldn't change anything in that case. It would return true and result in a patch being created, just as it does currently. Those checks for an "absolute value in disguise" are only meant for symbols defined using EQU/SET. They won't work with `@`.

Reply:
I see... And there's no way to pass constant offsets to the assembler without using `@` anyway.